lib/ecto/query/api.ex

```diff
@@ -469,9 +469,34 @@ defmodule Ecto.Query.API do
       field = "name"
       from(post in Post, select: post.meta["author"][^field])
 
+  See also `embed_extract_path/2` for working with embeds.
   """
   def json_extract_path(json_field, path), do: doc! [json_field, path]
 
+  @doc """
+  Returns value from the `embed_field` pointed to by `path`.
+
+      from(post in Post, select: json_extract_path(post.meta, [:author]))
+
+  The query can be also rewritten as:
+
+      from(post in Post, select: post.meta.author)
+
+  Path elements can be integers to access values in `embeds_many`:
+
+      from(post in Post, select: post.tags[0].name)
+
+  **Warning**: the underlying data in the JSON column is returned without
+  any additional decoding, e.g. datetimes (which are encoded as strings) are
+  returned as strings. This also means that queries like:
+  `where: post.meta.published_at > from_now(-1, "day")` may return incorrect results
+  as the underlying database may try to compare e.g. `json` with `date` types.
+  Use `type/2` to force the types on the database level.
```

Comment:
I know how to decode a value on retrieval (with `Ecto.Type.embedded_load/3`) but the problem in WHEREs would remain and that's more difficult to solve (e.g. on Postgres `(s0#>'{"date"}')::date` is not enough, we need to do `(s0#>'{"date"}')::text::date`). I think maybe we can ship this as is and revisit loading before shipping new release?

Reply:
Alternatively we can cast on load and not do anything on `where`, leaving it up to the database semantics. For example, if the database doesn't allow comparing jsonb with date, then you need to complain to the database. :)

Without casting/loading, I don't see much purpose on `embed_extract_path`. If the goal is to validate paths, then this can be done with `json_extract_path` itself. For example, if I write `p.embed["nested_embed"]["foo"]`, then I know "nested_embed" is an embed, so I can validate it exists or not. In other words, we can simply keep it `json_extract_path` and use the types that we know to validate it if possible. So the validation is based on the schema and not the syntax choice. Now I think this may even be preferrable. :) Thoughts?